packages/embark-blockchain-process/src/proxy.js

```diff
@@ -16,17 +16,26 @@ const modifyResponse = require('node-http-proxy-json');
 const Transaction = require('ethereumjs-tx');
 const ethUtil = require('ethereumjs-util');
 import { pingEndpoint } from './utils';
+import Web3 from 'web3';
 
-const METHODS_TO_MODIFY = {accounts: 'eth_accounts'};
+const METHODS_TO_MODIFY = {
+  accounts: 'eth_accounts',
+  personalAccounts: 'personal_listAccounts'
+};
 const REQUEST_TIMEOUT = 5000;
 
 const modifyPayload = (toModifyPayloads, body, accounts) => {
-  switch (toModifyPayloads[body.id]) {
-    case METHODS_TO_MODIFY.accounts:
-      delete toModifyPayloads[body.id];
-      body.result = Array.isArray(body.result) && body.result.concat(accounts);
-      break;
-    default:
+  if (toModifyPayloads[body.id] === METHODS_TO_MODIFY.accounts ||
+      toModifyPayloads[body.id] === METHODS_TO_MODIFY.personalAccounts) {
+    delete toModifyPayloads[body.id];
+    if (Array.isArray(body.result)) {
+      body.result = [
```

Comment:
You should make sure to check that result is an array with ` Array.isArray(body.result)` as before, because Geth sometimes returned not good stuff. It was a protection against bad outputs.

Reply:
I'm confused, isn't that what L31 does?

Reply:
Nah, you're not confused, it's me that's blind 🤦‍♂